coral-common/src/main/java/com/linkedin/coral/common/TypeConverter.java

```diff
@@ -141,11 +141,16 @@ public static RelDataType convert(StructTypeInfo structType, final RelDataTypeFa
   }
 
   // Mimic the StructTypeInfo conversion to convert a UnionTypeInfo to the corresponding RelDataType
+  // The schema of output Struct conforms to https://github.com/trinodb/trino/pull/3483
   public static RelDataType convert(UnionTypeInfo unionType, RelDataTypeFactory dtFactory) {
     List<RelDataType> fTypes = unionType.getAllUnionObjectTypeInfos().stream()
         .map(typeInfo -> convert(typeInfo, dtFactory)).collect(Collectors.toList());
-    List<String> fNames = IntStream.range(0, unionType.getAllUnionObjectTypeInfos().size()).mapToObj(i -> "tag_" + i)
+    List<String> fNames = IntStream.range(0, unionType.getAllUnionObjectTypeInfos().size()).mapToObj(i -> "field" + i)
         .collect(Collectors.toList());
+    if (fNames.size() > 0) {
+      fTypes.add(0, dtFactory.createSqlType(SqlTypeName.INTEGER));
```

Comment:
According to the corresponding PR:
https://github.com/trinodb/trino/pull/3483/files#diff-6a37b030f26bfeb6eca302ca157050cad2ead0476c1785fad57fc5f14fab88cfR264
the type of tag is `TINYINT`, rather than `INT`, so it should be 
```
fTypes.add(0, dtFactory.createSqlType(SqlTypeName.TINYINT));
```
querying the view will fail if the types mismatch.
I think the test also needs to be modified.

Enhanced coral-trino test suite helped me to find this hidden issue 😉

Reply:
Wow that's a great catch, thanks ! I will address this.

Reply:
Nice work for enhanced coral-trino test suite! This is awesome! We can catch more issues before hitting production.